src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurDetailLineGroupForCollector.java

```diff
@@ -156,14 +219,45 @@ protected KualiDecimal calculateTotalAmountForLines(List<ConcurStandardAccountin
     protected Optional<OriginEntryFull> buildOriginEntryForCashOffset(
             OriginEntryFull cashEntry, List<ConcurStandardAccountingExtractDetailLine> cashLines) {
         KualiDecimal cashAmount = getSignedAmountFromOriginEntry(cashEntry);
+        KualiDecimal cashAdvanceAmount = calculateTotalAmountForCashAdvanceLinesReferencedByRegularLines(cashLines);
+        KualiDecimal cashAmountToOffset = cashAmount.add(cashAdvanceAmount);
+        if (cashAmountToOffset.isZero()) {
+            return Optional.empty();

```

Comment:
Using an Optional allows for returning a wrapper around zero or one other objects, to avoid simply returning null for when a value does not exist. If this should just return null for simplicity, I could do that instead.